crates/starknet/src/transactions.rs

```diff
@@ -216,24 +216,28 @@ impl StarknetTransaction {
     pub fn get_l2_to_l1_messages(&self) -> Vec<MessageToL1> {
         let mut messages = vec![];
 
-        fn get_blockifier_messages_recursively(call_info: &CallInfo) -> Vec<OrderedMessageToL1> {
+        fn get_blockifier_messages_recursively(
+            call_info: &CallInfo,
+            caller_address: ContractAddress,
+        ) -> Vec<OrderedMessageToL1> {
             let mut messages = vec![];
 
-            let from_address = call_info.call.caller_address.into();
```

Comment:
I'm trying to understand the purpose of the extra argument (`caller_address`) you introduced to the recursion.
To me it looks like the removed line `let from_address = ...` could just be changed to:
```rust
let from_address = call_info.call.storage_address.into();
```
without the addition of the extra argument, and your PR would have the same effect.

Reply:
Will merge into a side branch and remove this recursive redundancy there. PR #296 is being updated. These two PRs are combined in #299.